modules/extension/xsd/xsd-fes/src/main/java/org/geotools/filter/v2_0/bindings/BBOXTypeBinding.java

```diff
@@ -52,4 +66,48 @@
     public QName getTarget() {
         return FES.BBOXType;
     }
+
+    public Class getType() {
+        return BBOX.class;
+    }
+
+    @Override
+    public Object getProperty(Object object, QName name) throws Exception {
+        BBOX box = (BBOX) object;
+        
+        if (FES.ValueReference.equals(name)) {
+            if (box.getExpression1() instanceof PropertyName) {
```

Comment:
Hmmm good point. I think the "if" can basically be removed without problems. This is a residue of older code that was altered. Will test with if removed.